platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy

```diff
@@ -17,13 +17,14 @@
 package org.gradle.internal.declarativedsl.settings
 
 import groovy.test.NotYetImplemented
-import org.gradle.api.internal.plugins.software.SoftwareType
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
-import org.gradle.test.fixtures.plugin.PluginBuilder
 import org.gradle.test.fixtures.server.http.MavenHttpPluginRepository
+import org.hamcrest.Matchers
 import org.junit.Rule
 
-class SoftwareTypeDeclarationIntegrationTest extends AbstractIntegrationSpec {
+import static org.gradle.util.Matchers.containsText
+
+class SoftwareTypeDeclarationIntegrationTest extends AbstractIntegrationSpec implements SoftwareTypeFixture {
```

Comment:
🤔 Other test ideas:
- What about if either model type is `private`?
- Multiple methods on a single plugin that each expose different software types?

Reply:
Both good ideas.  I've added some tests for these cases.